pages/docs/react/latest/styling.mdx

```diff
@@ -0,0 +1,207 @@
+---
+title: Styling
+description: "Styling in ReScript & React"
+canonical: "/docs/react/latest/styling"
+---
+
+# Styling
+
+There are many different approaches on how to do styling in React, such as:
+
+- Inline Styles (`style=...`)
+- Global CSS / CSS modules
+- CSS utility libraries (`tailwindcss`, `tachyons`, etc.)
+- CSS-in-JS (such as `styled-components`, `emotion`, etc.)
+
+There's no right or wrong when choosing a styling strategy, and each method has its up- and downsides. This page will give you some pointers on how to style your ReScript React components in your preferred style.
+
+
+## Inline Styles
+
+This is the most basic form of styling, coming straight from the 90ies. You can apply a `style` attribute to any DOM element with our `ReactDOM.Style.make` API:
+
+```rescript
+<div style=(
+  ReactDOM.Style.make(~color="#444444", ~fontSize="68px", ())
+)/>
+```
+
+It's a labeled (therefore typed) function call that maps to the familiar style object `{color: '#444444', fontSize: '68px'}`. For every CSS attribute in the CSS specfication, there is a camelCased label in our `make` function.
+
+**Note** that `make` returns an opaque `ReactDOM.Style.t` type that you can't read into. We also expose a `ReactDOM.Style.combine` that takes in two `style`s and combine them.
+
+### Escape Hatch: `unsafeAddProp`
+
+The above `Style.make` API will safely type check every style field! However, we might have missed some more esoteric fields. If that's the case, the type system will tell you that the field you're trying to add doesn't exist. To remediate this, we're exposing a `ReactDOM.Style.unsafeAddProp` to dangerously add a field to a style:
+
+```reason
+let myStyle = ReactDOM.Style.make(~color="#444444", ~fontSize="68px", ());
+```
+
+## Global CSS
+
+Use a `%%raw` expression to import CSS files within your ReScript / React component code:
+
+```rescript
+// in a CommonJS setup
+%%raw("require('./styles/main.css')")
+
+// or with ES6
+%%raw("import './styles/main.css'")
+```
+
+## CSS Modules
+
+[CSS modules](https://github.com/css-modules/css-modules) can be imported like any other JS module. The imported value is a JS object, with attributes equivalent to each classname defined in the CSS file.
+
+As an example, let's say we have a CSS module like this:
+
+```css
+/* styles.module.css */
+
+.root {
+  color: red
+}
+```
+
+We now need to create a module binding that imports our styles as a JS object: 
+
+```res
+// {..} means we are handling a JS object with an unknown
+// set of attributes
+@module external styles: {..} = "./styles.module.css"
+
+// Use the obj["key"] syntax to access any classname within our object
+let app = <div className={styles["root"]} />
+```
+
+**Note:** `{..}` is an open JS object type, which means the type checker will not type check correct classname usage. If you want to enforce compiler errors, replace `{..}` with a concrete JS object type, such as `{"root": string}`.
+
+
+## CSS Utility Libraries
+
```

Comment:
I definitely like the idea of showing that using common css libs is straightforward but I also think that just posting these two might show like a super strong endorsement when really they are some examples that we have used recently. Maybe a small disclaimer that there are other options out there as well?

```suggestion

There are a number of third party libraries for styling React components and you might be comfortable with a specific setup. If they work in JS then they almost certainly work in ReScript. Below, we've shared some ideas for working with popular libraries. If you're interested in working with one you don't see here, search the [package index](https://rescript-lang.org/packages) or post in [the forum](https://forum.rescript-lang.org)

```